crates/host/src/wasmbus/mod.rs

```diff
@@ -2086,6 +2102,774 @@ impl Host {
+                                    }
+                                }
+                                else {
+                                    warn!(?provider_id, "failed to request provider health, retrying in 30 seconds");
+                                }
+                        }
+                        exit = exit_health_rx.recv() => {
+                            if let Err(err) = exit {
+                                warn!(%err, ?provider_id, "failed to receive exit in health check task");
+                            }
+                            break;
+                        }
+                    }
+                }
+            }
+        });
+
+        // Spawn off a task to watch for config bundle updates and forward them to
+        // the provider that we're spawning and managing
+        let mut exit_config_tx = exit_rx;
+        let provider_id = Arc::clone(&provider_id);
+        let lattice = Arc::clone(&self.host_config.lattice);
+        tasks.spawn({
+             let config = Arc::clone(&config);
+             async move {
+                 let subject = provider_config_update_subject(&lattice, &provider_id);
+                 trace!(?provider_id, "starting config update listener");
+                 loop {
+                     let mut config = config.write().await;
+                     select! {
+                         maybe_update = config.changed() => {
+                             let Ok(update) = maybe_update else {
+                                 break;
+                             };
+                             trace!(?provider_id, "provider config bundle changed");
+                             let bytes = match serde_json::to_vec(&*update) {
+                                 Ok(bytes) => bytes,
+                                 Err(err) => {
+                                     error!(%err, ?provider_id, ?lattice, "failed to serialize configuration update ");
+                                     continue;
+                                 }
+                             };
+                             trace!(?provider_id, subject, "publishing config bundle bytes");
+                             if let Err(err) = rpc_nats.publish(subject.clone(), Bytes::from(bytes)).await {
+                                 error!(%err, ?provider_id, ?lattice, "failed to publish configuration update bytes to component");
+                             }
+                         }
+                         exit = exit_config_tx.recv() => {
+                             if let Err(err) = exit {
+                                 warn!(%err, ?provider_id, "failed to receive exit in config update task");
+                             }
+                             break;
+                         }
+                     }
+                 }
+             }
+         });
+        Ok(())
+    }
+
+    #[instrument(level = "debug", skip_all)]
+    async fn start_http_provider(
+        &self,
```

Comment:
refactoring nit that can be done as a followup: Any chance we could move these to separate `impl` blocks in separate modules? Since there's a whole provider implementation for http and messaging here, this is extending an already large wasmbus/mod.rs

Reply:
done